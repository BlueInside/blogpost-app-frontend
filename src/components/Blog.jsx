// import { useEffect } from 'react';
import PropTypes from 'prop-types';
import usePosts from './hooks/usePosts';
function Post({ title, content }) {
  return (
    <div>
      <h2>{title}</h2>
      <div>
        <p>{content}</p>
      </div>
    </div>
  );
}

Post.propTypes = {
  id: PropTypes.string,
  title: PropTypes.string,
  content: PropTypes.string,
};

function Blog() {
  // Use custom hook to fetch posts
  const { data: posts, error, loading } = usePosts();

  if (loading) return <div>Loading...</div>;
  if (error) return <p>{error.message}</p>;

  return (
    <>
      {posts &&
        posts.map((post) => (
          <Post
            key={post.id}
            id={post.id}
            title={post.title}
            content={post.content}
          />
        ))}
    </>
  );
}

export default Blog;
