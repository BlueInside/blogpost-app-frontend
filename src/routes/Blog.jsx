// import { useEffect } from 'react';
import PropTypes from 'prop-types';
import usePosts from '../components/hooks/usePosts';
import { useNavigate } from 'react-router-dom';

function Post({ id, title, content }) {
  const navigate = useNavigate();

  return (
    <article
      role="post"
      onClick={() => {
        navigate(`/post/${id}`);
      }}
    >
      <h2>{title}</h2>
      <div>
        <p>{content}</p>
      </div>
    </article>
  );
}

Post.propTypes = {
  id: PropTypes.string,
  title: PropTypes.string,
  content: PropTypes.string,
};

function Blog() {
  // Use custom hook to fetch posts
  const {
    data: posts,
    error,
    loading,
  } = usePosts('http://localhost:3000/posts');

  if (loading) return <div>Loading...</div>;
  if (error) return <p>{error.message}</p>;

  return (
    <>
      {posts &&
        posts.map((post) => (
          <Post
            key={post._id}
            id={post._id}
            title={post.title}
            content={post.content}
          />
        ))}
    </>
  );
}

export default Blog;
