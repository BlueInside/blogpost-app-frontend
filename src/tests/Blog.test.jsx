import { describe, it, vi } from 'vitest';
import Blog from '../routes/Blog';
import { render, screen } from '@testing-library/react';
import { expect } from 'chai';

let fakeFetchData = {
  data: [
    { _id: '1', title: 'Post 1', content: 'Text of post 1' },
    { _id: '2', title: 'Post 2', content: 'Text of post 2' },
  ],
};

describe('Blog component', () => {
  vi.mock('../components/hooks/usePosts', () => {
    return {
      default: () => fakeFetchData,
    };
  });

  it('Renders blog component properly', () => {
    const { container } = render(<Blog />);
    expect(container).toMatchSnapshot();
  });

  it('Renders posts correctly', () => {
    render(<Blog />);
    expect(screen.getAllByRole('heading')).toHaveLength(2);
  });
});
