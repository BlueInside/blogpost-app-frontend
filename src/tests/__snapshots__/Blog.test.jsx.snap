// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`__default_name_ 1`] = `
<div>
  <article
    role="post"
  >
    <h2>
      Post 1
    </h2>
    <div>
      <p>
        Text of post 1
      </p>
    </div>
  </article>
  <article
    role="post"
  >
    <h2>
      Post 2
    </h2>
    <div>
      <p>
        Text of post 2
      </p>
    </div>
  </article>
</div>
`;
